src/iidm/util/Substations.cpp

```diff
@@ -41,28 +41,30 @@ void checkRemovability(const Substation& substation) {
 }
 
 void checkRemovability(const Substation& substation, const Branch& branch) {
-    const Substation& s1 = branch.getTerminal1().getVoltageLevel().getSubstation();
-    const Substation& s2 = branch.getTerminal2().getVoltageLevel().getSubstation();
-    if (!stdcxx::areSame(s1, substation) || !stdcxx::areSame(s2, substation)) {
+    const auto& s1 = branch.getTerminal1().getVoltageLevel().getSubstation();
+    const auto& s2 = branch.getTerminal2().getVoltageLevel().getSubstation();
+    if ((static_cast<bool>(s1) && !stdcxx::areSame(s1.get(), substation)) || (static_cast<bool>(s2) && !stdcxx::areSame(s2.get(), substation))) {
```

Comment:
It could be interesting to extract the test `(static_cast<bool>(s1) && !stdcxx::areSame(s1.get(), substation))` in a utility function

Reply:
Agree. I added it in Substations.cpp file.
```
bool areDifferentSubstations(const stdcxx::CReference<Substation>& s1, const Substation& s2) {
    return static_cast<bool>(s1) && !stdcxx::areSame(s1.get(), s2);
}
```

I could hav added in memory.hpp, because this function is not intuitive enough IMHO : I would expect it returns true if the first parameter is invalid but it does not => to be discussed